docker_test.go

```diff
@@ -46,6 +50,65 @@ func init() {
 	}
 }
 
+func TestWithinContainerStage1(t *testing.T) {
+	// TODO: remove this skip check when context rework is merged.
+	if providerType != ProviderDocker {
+		t.Skip("This is a docker-specific test.")
+	}
+	dir, err := os.Getwd()
+	require.NoError(t, err)
+	req := ContainerRequest{
+		Image:      golangImage,
+		WorkingDir: dir,
+		HostConfigModifier: func(hc *container.HostConfig) {
+			hc.Mounts = append(hc.Mounts, mount.Mount{
+				Type:   "bind",
+				Source: dir,
+				Target: dir,
+			}, mount.Mount{
+				Type:   "bind",
+				Source: "/var/run",
+				Target: "/var/run",
+			})
+		},
+		Entrypoint: []string{"sleep", "100"},
+	}
```

Comment:
If what we want to do is run this in container in rootless mode, we'd need to set up the environment for that. You can find the rootless paths in the [docker_rootless.go:L35:39](https://github.com/testcontainers/testcontainers-go/blob/main/internal/core/docker_rootless.go#L35-L39):

> //  1. XDG_RUNTIME_DIR environment variable.
> //  2. ~/.docker/run/docker.sock file.
> //  3. ~/.docker/desktop/docker.sock file.
> //  4. /run/user/${uid}/docker.sock file.
> //  5. Else, return ErrRootlessDockerNotFound, wrapping secific errors for each of the above paths.

If you need to run this test in both modes, you probably need to configure the image accordingly. Please see https://github.com/testcontainers/testcontainers-go/blob/main/docker_test.go#L1546-L1548 as an example

```go
if os.Getenv("XDG_RUNTIME_DIR") != "" {
	t.Skip("Rootless Docker does not support setting rlimit")
}
```

Reply:
I pushed an attempt to fix the test, but you'll have to approve the CI run to find out if it worked. Feel free to take over my PR from here to fix the docker socket mounting issue if it's easier. If this doesn't work, it might be necessary to pass into the container XDG_RUNTIME_DIR.

Reply:
I pushed one last attempt to fix the test. Please take over from here if it doesn't work.